### ISSUE TITLE
text-align: justify is not incremental layout safe

### ISSUE THREAD

**Issue by mbrubeck** [servo/servo#6502]
When the `text-align` property is set to `justify`, `InlineFlow::justify_inline_fragments` adds extra spacing to the `advance` values in the glyph stores of the affected text fragments.  This extra spacing is not removed if the property value changes, and can even be added again each time incremental layout runs, causing the spacing to increase and eventually overflow.

You can see this by repeatedly mousing in and out of the text in this example:

```html
<html>
  <head>
    <style>
      p { width: 15em; }
      p:hover { text-align: justify; border: 1px solid red; }
    </style>
  </head>
  <body>
    <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
    tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
    veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea
    commodo consequat. Duis aute irure dolor in reprehenderit in voluptate
    velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat
    cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id
    est laborum.</p>
  </body>
</html>
```

This is similar to #6501.  cc @pcwalton

**Comment by eefriedman** [comment]
This appears to work correctly on master.

**Comment by mbrubeck** [comment]
Possibly fixed by #7821.
